tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -68,37 +69,43 @@ def __init__(self, name, init_config, instances):
                         self.CONFIGURATION_ERROR_MSG.format(self.host, 'IPv6 address', 'valid address')
                     )
             # It's a correct IP V6 address
-            self._addr = self.host
+            self._addrs = self.host
             self.socket_type = socket.AF_INET6
         else:
             self.socket_type = socket.AF_INET
             # IP will be resolved at check time
 
     @property
-    def addr(self):
-        if self._addr is None:
+    def addrs(self):
+        if self._addrs is None or self._addrs == []:
             try:
                 self.resolve_ip()
             except Exception as e:
                 self.log.error(str(e))
                 msg = "URL: {} could not be resolved".format(self.host)
                 raise CheckException(msg)
-        return self._addr
+        return self._addrs
 
     def resolve_ip(self):
-        self._addr = socket.gethostbyname(self.host)
-        self.log.debug("%s resolved to %s", self.host, self._addr)
+        if self.multiple_ips:
+            _, _, self._addrs = socket.gethostbyname_ex(self.host)
+        else:
+            self._addrs = [socket.gethostbyname(self.host)]
+
+        if self._addrs == []:
+            raise Exception("No IPs attached to host")
```

Comment:
What do you think about also submitting a service check here?

Reply:
I think it's a good idea but of out scope of this PR, since we didn't submit critical service check on resolve ip failure before

Reply:
Sounds good, makes sense